source/jormungandr/jormungandr/georef.py

```diff
@@ -98,13 +100,33 @@ def get_crow_fly(
         req.places_nearby.start_page = 0
         req.disable_feedpublisher = True
         req.places_nearby.types.append(object_type)
+
+        allowed_id_filter = ''
+        if allowed_id is not None:
+            allowed_id_count = len(allowed_id)
+            if allowed_id_count > 0:
+                # Retrieve first element(uri) of the set
+                allowed_id_items = 'poi.id={}'.format(allowed_id.pop())
+                # For the rest of elements use 'or' in between two elements(uris)
+                for uri in allowed_id:
+                    allowed_id_items = allowed_id_items + ' or poi.id={}'.format(uri)
+                # Format the filter for all allowed_ids uris
+                if allowed_id_count == 1:
+                    allowed_id_filter = ' and {}'.format(allowed_id_items)
+                elif allowed_id_count > 1:
+                    allowed_id_filter = ' and ({})'.format(allowed_id_items)
+
+        # We implement filter only for poi with poi_type.uri=poi_type:amenity:parking
         if filter is not None:
-            req.places_nearby.filter = filter
+            req.places_nearby.filter = filter + allowed_id_filter
         if streetnetwork_mode == "car":
             req.places_nearby.stop_points_nearby_radius = (
                 kwargs.get("walking", 1.11) * stop_points_nearby_duration
             )
             req.places_nearby.depth = 1
+        if forbidden_uris is not None:
+            for uri in forbidden_uris:
+                req.places_nearby.forbidden_uris.append(uri)
```

Comment:
Why can't we do the same for `allowed_ids` ? Is there a missing interface in PTref ? we should definitely add it to PtRef if it's not there. Otherwise, we will duplicate the same business logic everywhere we need allowed ids.

Reply:
allowed_ids are only used by JourneyRequest (api journeys) but not in other endpoints. https://github.com/CanalTP/navitia-proto/blob/85ae96e65a1b54764cb1f0c614353909eed51a48/request.proto#L87